### PR TITLE
feat: Add archipelago support

### LINKED ISSUE #2404: Add support for archipelago
- [x] Integrate new peer library
- [x] Allow to pick only catalyst (not layer)
- [x] Add url parameter for island
- [x] Forward island url parameter if provided to peer library for prefered island
- [x] Fix "hot scenes" algorithm, adding support for both old & new lighthouses
- [x] Review /changerealm. Maybe support changeisland? &amp; Maybe support changeserver?

### PR DESCRIPTION
**Pull request by pablitar**
This PR adds archipelago support to explorer. For a while, explorer will support both kinds of realms: Layer Based Realms (the current ones) and Archipelago realms (the new ones).

We need to merge this into master in order to be able to deploy archipelago to Catalysts that are in DAO. Otherwise, explorer will crash when querying the DAO. This will also need to be released before any production catalyst is updated to archipelago too.

On Archipelago Realms, layers will be removed and replace them with "islands" that get dynamically generated. Their name will now be catalyst-name-only (for instance, `fenrir`)

A prefered island can be provided through URL in order for friends to be able to connect when multiple islands intersect.

Closes #2404. Will be followed by #2488.

### CODE REVIEW

**Comment by pablitar** [comment]
https://play.decentraland.zone/branch/feat/add-archipelago/index.html?ENV=zone&CATALYST=https://peer-uw-1.decentraland.zone/ <- To test with a catalyst that is not in DAO, and that has archipelago support

File: kernel/packages/shared/comms/index.ts [hunk 1]
```diff
@@ -1133,7 +1145,7 @@ async function doStartCommunications(context: Context) {
         obj.immediate
       ] as Position
 
-      if (context && isRendererEnabled) {
+      if (context && isRendererEnabled()) {
```
Comment (menduz) [inline kernel/packages/shared/comms/index.ts:1149]: Oh dear lord of JS

File: kernel/package.json [hunk 2]
```diff
@@ -81,7 +81,7 @@
     "node-fetch": "^2.6.1",
     "nyc": "^14.1.1",
     "prettier": "^2.0.5",
-    "query-string": "^5.0.1",
+    "query-string": "^7.0.1",
```
Comment (pablitar) [inline kernel/package.json:84]: This library has been updated because it was untyped

File: kernel/packages/shared/dao/index.ts [hunk 3]
```diff
@@ -41,6 +52,17 @@ const score = ({ usersCount, maxUsers = 50 }: Layer) => {
   return v + v * Math.cos(phase + period * usersCount)
 }
 
+// When using islands, the more users the better. We also prioritize these over the old lighthouses.
+// That means the score starts over 100 for any candidate that has at least 1 user.
+// We should probably revamp score altogether
```
Comment (pablitar) [inline kernel/packages/shared/dao/index.ts:57]: This will be revamped in #2488

File: kernel/packages/shared/chat/sagas.ts [hunk 4]
```diff
@@ -487,12 +488,11 @@ function initChatCommands() {
...
-            sceneInfo.name.length < 20 ? sceneInfo.name : sceneInfo.name.substring(0, 20) + '...'
-          } ${sceneInfo.baseCoords.x},${sceneInfo.baseCoords.y} ${sceneInfo.realms.reduce(
-            (a, b) => a + `\n\t realm: ${b.serverName}-${b.layer} users: ${b.usersCount}`,
-            ''
-          )}\n`
+          body += `${count} ${count > 1 ? 'users' : 'user'} @ ${sceneInfo.name.length < 20 ? sceneInfo.name : sceneInfo.name.substring(0, 20) + '...'
+            } ${sceneInfo.baseCoords.x},${sceneInfo.baseCoords.y} ${sceneInfo.realms.reduce(
+              (a, b) => a + `\n\t realm: ${b.serverName}-${b.layer} users: ${b.usersCount}`,
+              ''
+            )}\n`
```
Comment (menduz) [inline kernel/packages/shared/chat/sagas.ts]: I believe in code beauty.

```
const parts = []

parts.push(count)
parts.push(count)
if (something) {
  parts.push(count)
}

body += parts.join(' ')
```

Not a change request. Personal appreciation

File: kernel/packages/shared/comms/index.ts [hunk 5]
```diff
@@ -1287,3 +1300,32 @@ function stripSnapshots(profile: Profile): Profile {
...
+    const q = qs.parse(location.search)
+
+    if (currentIsland) {
+      q.island = currentIsland
+    } else {
+      delete q.island
+    }
+
+    history.replaceState({ island: currentIsland }, '', `?${qs.stringify(q)}`)
+  })
```
Comment (menduz) [inline kernel/packages/shared/comms/index.ts:1330]: I love the observable tied to the store.

File: kernel/packages/shared/chat/sagas.ts [hunk 6]
```diff
@@ -487,12 +488,11 @@ function initChatCommands() {
...
           const count = sceneInfo.realms.reduce((a, b) => a + b.usersCount, 0)
-          body += `${count} ${count > 1 ? 'users' : 'user'} @ ${
-            sceneInfo.name.length < 20 ? sceneInfo.name : sceneInfo.name.substring(0, 20) + '...'
-          } ${sceneInfo.baseCoords.x},${sceneInfo.baseCoords.y} ${sceneInfo.realms.reduce(
-            (a, b) => a + `\n\t realm: ${b.serverName}-${b.layer} users: ${b.usersCount}`,
-            ''
-          )}\n`
+          body += `${count} ${count > 1 ? 'users' : 'user'} @ ${sceneInfo.name.length < 20 ? sceneInfo.name : sceneInfo.name.substring(0, 20) + '...'
+            } ${sceneInfo.baseCoords.x},${sceneInfo.baseCoords.y} ${sceneInfo.realms.reduce(
+              (a, b) => a + `\n\t realm: ${b.serverName}-${b.layer} users: ${b.usersCount}`,
```
Comment (nchamo) [inline kernel/packages/shared/chat/sagas.ts]: Should we also replace this part `{b.serverName}-${b.layer}`?

**Review by nchamo — APPROVED** [review]
Great work!

File: kernel/packages/shared/chat/sagas.ts (hunk 6 above)
Comment (pablitar) [reply kernel/packages/shared/chat/sagas.ts]: Yep. We probably should.

**Review by menduz — COMMENTED** [review]
Sorry. this review remained in draft. Dismiss my comments.

**Pull request merged**